packages/mercury/src/components/paginator/_paginator-styles.scss

```diff
@@ -0,0 +1,271 @@
+/// @param {String} $paginator__item--disabled-selector [".paginator::part(item disabled)"] -
+/// @param {String} $paginator__item--nested-selector [".paginator::part(item nested)"] -
+/// @param {String} $paginator__item--selected-selector [".paginator::part(item selected)] -
+/// @param {Boolean} $add--disabled [false] -
+/// @param {Boolean} $add--placeholder [false] -
+/// @param {Boolean} $add--error [false] -
+@mixin paginator(
+  $paginator-selector: ".paginator",
+  $paginator__go-to-selector: ".paginator::part(go-to)",
+  $paginator__go-to__input-selector: ".paginator::part(go-to__input)",
+  $paginator__items-per-page-selector: ".paginator::part(items-per-page)",
+  $paginator__items-per-page__label-selector:
+    ".paginator::part(items-per-page__label)",
+  $paginator__items-per-page__info-selector:
+    ".paginator::part(items-per-page-info)",
+  $paginator__navigation-page-selector: ".paginator::part(page)",
+  $paginator__first-button-selector: ".paginator::part(first__button)",
+  $paginator__last-button-selector: ".paginator::part(last__button)",
+  $paginator__next-button-selector: ".paginator::part(next__button)",
+  $paginator__previous-button-selector: ".paginator::part(prev__button)",
+
+  $paginator__combo-box-selector: ".paginator::part(items-per-page__combo-box)",
+  $paginator__combo-box--disabled-selector: ".paginator[disabled]",
+  $paginator__combo-box__placeholder-selector:
+    ".paginator[part='ch-combo-box-render--placeholder']",
+  $paginator__window-selector: ".paginator::part(window)",
+  $paginator__group-selector: ".paginator::part(group)",
+  $paginator__group__content-selector: ".paginator::part(group__content)",
+  $paginator__group__header-selector: ".paginator::part(group__header)",
+  $paginator__group__header--disabled-selector:
+    ".paginator::part(group__header disabled)",
+  $paginator__group__header--expandable-selector:
+    ".paginator::part(group__header expandable)",
+  $paginator__item-selector: ".paginator::part(item)",
+  $paginator__item--disabled-selector: ".paginator::part(item disabled)",
+  $paginator__item--nested-selector: ".paginator::part(item nested)",
+  $paginator__item--selected-selector: ".paginator::part(item selected)",
+  $add--disabled: false,
+  $add--placeholder: false,
+  $add--error: false
+) {
+  #{$paginator-selector} {
+    @extend %paginator;
+  }
+
+  #{$paginator__go-to-selector} {
+    @extend %paginator__go-to;
+  }
+
+  #{$paginator__items-per-page__info-selector} {
+    @extend %paginator__items-per-page__info;
+  }
+
+  #{$paginator__navigation-page-selector} {
+    @extend %paginator__button;
+
+    &:hover {
+      @extend %paginator__button--hover;
+    }
+  }
+
+  #{$paginator__previous-button-selector} {
+    @extend %paginator__button;
+
+    &::before {
+      @extend %paginator__button__selector--before;
+      --button-arrow-icon: var(--icon__system_chevron-left_primary--enabled);
```

Comment:
If possible, apply the css variable inside the `@extend`, just like the other styles.